samples/notebooks/dotnet/09-BingSearch-using-kernel-prince.ipynb

```diff
@@ -0,0 +1,320 @@
+    "\n",
+    "var questions = \"Who is the most followed person on TikTok right now? What's the exchange rate EUR:USD?\";\n",
+    "Console.WriteLine(questions);\n",
+    "\n",
+    "var oracle = kernel.CreateSemanticFunction(SemanticFunction, maxTokens: 200, temperature: 0, topP: 1);\n",
+    "var context = kernel.CreateNewContext();\n",
+    "context[\"externalInformation\"] = \"\";\n",
+    "var answer = await oracle.InvokeAsync(questions, context);\n",
+    "\n",
+    "// If the answer contains commands, execute them using the prompt renderer.\n",
+    "if (answer.Result.Contains(\"bing.search\", StringComparison.OrdinalIgnoreCase))\n",
+    "        {\n",
+    "            var promptRenderer = new PromptTemplateEngine();\n",
+    "\n",
+    "            Console.WriteLine(\"---- Fetching information from Bing...\");\n",
+    "            var information = await promptRenderer.RenderAsync(answer.Result, context);\n",
+    "\n",
+    "            Console.WriteLine(\"Information found:\");\n",
+    "            Console.WriteLine(information);\n",
+    "\n",
+    "            // The rendered prompt contains the information retrieved from search engines\n",
+    "            context[\"externalInformation\"] = information;\n",
+    "\n",
+    "            // Run the semantic function again, now including information from Bing\n",
+    "            answer = await oracle.InvokeAsync(questions, context);\n",
+    "        }\n",
+    "        else\n",
+    "        {\n",
+    "            Console.WriteLine(\"AI had all the information, no need to query Bing.\");\n",
+    "        }\n",
+    "\n",
+    "Console.WriteLine(\"---- ANSWER:\");\n",
+    "Console.WriteLine(answer);\n",
+    "\n",
+    "        /* OUTPUT:\n",
+    "\n",
+    "            Who is the most followed person on TikTok right now? What's the exchange rate EUR:USD?\n",
+    "            ---- Fetching information from Bing...\n",
+    "            Information found:\n",
+    "\n",
+    "            Khaby Lame is the most-followed user on TikTok. This list contains the top 50 accounts by number\n",
+    "            of followers on the Chinese social media platform TikTok, which was merged with musical.ly in 2018.\n",
+    "            [1] The most-followed individual on the platform is Khaby Lame, with over 153 million followers..\n",
+    "            EUR – Euro To USD – US Dollar 1.00 Euro = 1.10 37097 US Dollars 1 USD = 0.906035 EUR We use the\n",
+    "            mid-market rate for our Converter. This is for informational purposes only. You won’t receive this\n",
+    "            rate when sending money. Check send rates Convert Euro to US Dollar Convert US Dollar to Euro..\n",
+    "            ---- ANSWER:\n",
+    "\n",
+    "            * The most followed person on TikTok right now is Khaby Lame, with over 153 million followers.\n",
+    "            * The exchange rate for EUR to USD is 1.1037097 US Dollars for 1 Euro.\n",
+    "         */\n",
+    "}"
+   ]
```

Comment:
This example, as well as example 1 above, appears to be identical to what is in [Example07_BingAndGoogleSkills.cs](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/KernelSyntaxExamples/Example07_BingAndGoogleSkills.cs). Rather than just repeat the code here to be run in the notebook, we want to use the notebooks to walk users step by step through what they are doing and explain each step along the way. Could you break these code chunks into smaller steps and add some explanation in between?

Reply:
I have removed this example

Reply:
I have removed this example